src/commands/deploy/nextjs/deploy.ts

```diff
@@ -39,22 +42,43 @@ import { computeAssetsPaths } from "./assets.js";
 import * as Sentry from "@sentry/node";
 import { randomUUID } from "crypto";
 import { EdgeFunction, getEdgeFunctions } from "./edge.js";
-import { uploadEnvVarsFromFile, uploadUserCode } from "../utils.js";
+import { attemptToInstallDependencies, uploadEnvVarsFromFile, uploadUserCode } from "../utils.js";
 import { readOrAskConfig } from "../utils.js";
 
+type NextJsComponent = {
+    path: string;
+    packageManager: PackageManagerType;
+    scripts?: {
+        deploy: string;
+        build?: string;
+        start?: string;
+    };
+    environment?: {
+        [key: string]: string;
+    };
+    subdomain?: string;
+};
```

Comment:
Looks like this is more of a SSRFramework, than specific to NextJS

Reply:
You are right, I have updated it and move it on a higher level.